src/util/sharing_map.h

```diff
@@ -39,6 +39,66 @@ Author: Daniel Poetzl
   CV typename sharing_mapt<keyT, valueT, hashT, predT>::ST \
   sharing_mapt<keyT, valueT, hashT, predT>
 
+/// A map implemented as a tree where subtrees can be shared between different
+/// maps.
```

Comment:
Can you somewhere explain how to generate sharing (likely via copy and/or assignment operations)?

Reply:
Yes, I'll add an explanation for that.